File: romantoInt.js
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function(s) {
    symbolDict = {'I': 1, 'V': 5, 'X':10, 'L':50, 'C':100, 'D':500, 'M':1000};
    var len = s.length;
    var last = '';
    var res = 0;
    for(var i = len - 1; i >= 0; i--){
        var cur = s.charAt(i);
        if(symbolDict[cur] < last){
           res = res - symbolDict[cur]; 
        } 
        else{
           res += symbolDict[cur]; 
        } 
        last = symbolDict[cur];
    }
    return res;
};

File: romantoInt.py
class Solution:
    def romanToInt(self, s: str) -> int:
        romanDict = {'I':1, 'V':5,'X':10,'L':50,'C':100,'D':500,'M':1000}
        res = 0
        for i in range(len(s)):
            cur = romanDict[s[i]]
            last = 0
            if i >= 1: last = romanDict[s[i-1]]
            print(i, cur, last)
            if cur > last:
                res += cur - last*2
            else:
                res += cur
        return res